src/main/java/rx/Completable.java

```diff
@@ -1971,8 +1972,29 @@ public final void subscribe(CompletableSubscriber s) {
         } catch (NullPointerException ex) {
             throw ex;
         } catch (Throwable ex) {
+            Exceptions.throwIfFatal(ex);
             ERROR_HANDLER.handleError(ex);
+            throw toNpe(ex);
+        }
+    }
+
+    /**
+     * Subscribes the given CompletableSubscriber to this Completable instance
+     * and handles exceptions throw by its onXXX methods.
+     * @param s the CompletableSubscriber, not null
+     * @throws NullPointerException if s is null
+     */
+    public final void safeSubscribe(CompletableSubscriber s) {
+        requireNonNull(s);
+        try {
+            // TODO plugin wrapping the subscriber
```

Comment:
The plugin system is half mistery to me right. I'll figure that out and add the hooks in a separate PR.